src/Microsoft.AspNet.Razor/Parser/TagHelpers/TagHelperBlockRewriter.cs

```diff
@@ -134,11 +137,14 @@ public static class TagHelperBlockRewriter
                     SourceLocation symbolStartLocation;
 
                     // Check for attribute start values, aka single or double quote
-                    if (IsQuote(htmlSymbols[i + 1]))
+                    if (i + 1 < htmlSymbols.Length && IsQuote(htmlSymbols[i + 1]))
```

Comment:
this condition isn't in the right spot because whitespace is allowed between the equals sign and the opening quote or value.  IOW this method needs more states in its machine or to make existing states easier to see:
1. before name (skip whitespace if preceding code hasn't already done that)
2. name found (now skip whitespace)
3. equals found (now skip whitespace)
4. bare or quoted value found (perhaps need to skip whitespace after end of value or the matching quote

side question which branches in this method will `checked`, `@multiple`, `type = @type`, and `type = "@type"` hit?  realize `@multiple` isn't currently supported but I'm curious nonetheless.